src/main/kotlin/thedarkcolour/kotlinforforge/forge/KDeferredRegister.kt

```diff
@@ -30,18 +29,12 @@ public typealias KDeferredRegister<T> = DeferredRegister<T>
  * @param V Type of deferred register
  * @param T Specific type of object being registered
  */
-public inline fun <V, T : V> KDeferredRegister<V>.registerObject(
+public fun <V, T : V> KDeferredRegister<V>.registerObject(
     name: String,
-    noinline supplier: () -> T
-): ReadOnlyProperty<Any?, T> {
+    supplier: () -> T
+): ObjectHolderDelegate<T> {
     val registryObject = this.register(name, supplier)
 
     // note that this anonymous class inherits three types
```

Comment:
This should be removed too now that there is a concrete type

Reply:
KDeferredRegister.kt is now reverted